### PR TITLE
analysis: try to detect when one of the worker was killed by OOM killer

### PR DESCRIPTION
**Pull request by tomato42**
### Description

When one of the worker processes is killed by an OOM killer, the `multiprocess` module doesn't handle it gracefully (doesn't reissue task), which makes the whole analysis hang. Try to detect that situation.

### Motivation and Context

Make error reporting in multiprocess execution more robust.

### Checklist

- [x] I have read the [CONTRIBUTING.md](https://github.com/tlsfuzzer/tlsfuzzer/blob/master/CONTRIBUTING.md) document and my PR follows [change requirements](https://github.com/tlsfuzzer/tlsfuzzer/blob/master/CONTRIBUTING.md#change-requirements) therein
- [x] the changes are also reflected in documentation and code comments
- [ ] all new and existing tests pass (see CI results)
- [x] [test script checklist](https://github.com/tlsfuzzer/tlsfuzzer/wiki/Test-script-checklist) was followed for new scripts
- [x] new test script added to `tlslite-ng.json` and `tlslite-ng-random-subset.json`
- [x] new and modified scripts were ran against popular TLS implementations:
  - n/a
- [x] required version of tlslite-ng updated in requirements.txt and README.md

- - -
This change is [Reviewable](https://reviewable.io/reviews/tlsfuzzer/tlsfuzzer/980)

### CODE REVIEW

File: tlsfuzzer/analysis.py [hunk 1]
```diff
@@ -526,10 +526,20 @@ def mt_process(self, sum_func, args=()):
...
                 self._mt_process_runner,
                 zip(comb, repeat(sum_func), repeat(args)),
-                job_size))
+                job_size
+            ):
+                pvals.append(i)
+
+                workers.update(pool._pool)
+                if any(not p.is_alive() for p in workers):
+                    raise RuntimeError("One of workers killed")
```
Comment (GeorgePantelakis) [inline tlsfuzzer/analysis.py:541]: nit: "One of the workers was killed"? Also, maybe I would like to see more info on the Runtime Error, perhaps the exit code of the process.

**Review by GeorgePantelakis — APPROVED** [review]
LGTM, only one nit found. It is kind of tricky to access a "private" variable of the pools but I don't see any other way.

File: tlsfuzzer/analysis.py (hunk 1 above)
Comment (GeorgePantelakis) [reply tlsfuzzer/analysis.py:541]: Exit code could be kind of useful, https://docs.python.org/3/library/multiprocessing.html#multiprocessing.Process.exitcode

Comment (tomato42) [reply tlsfuzzer/analysis.py:546]: exit code shouldn't matter too much, but I've updated it to print status like this:
```
RuntimeError: One of the workers was killed: <ForkProcess name='ForkPoolWorker-5:7' pid=38822 parent=38814 stopped exitcode=-SIGSEGV daemon>
```

**Review by GeorgePantelakis — APPROVED** [review]
LGTM

**Pull request merged**